docs/sources/flow/reference/components/pprof.scrape.md

```diff
@@ -0,0 +1,246 @@
+
+### http_client_config block
+
+The `http_client_config` block configures settings used to connect to
+endpoints.
+
+{{< docs/shared lookup="flow/reference/components/http-client-config-block.md" source="agent" >}}
+
+### basic_auth block
+
+{{< docs/shared lookup="flow/reference/components/basic-auth-block.md" source="agent" >}}
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### oauth2 block
+
+{{< docs/shared lookup="flow/reference/components/oauth2-block.md" source="agent" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+### profiling_config block
+
+The `profiling_config` block configures the profiling settings when scraping targets.
+
+The block contains the following attributes:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`path_prefix`                  | `string`     | The path prefix to use when scraping targets. | | no
+`pprof_config`               | `map(PprofProfilingConfig)` | The pprof profile endpoints configuration. | see below | no
+
+By default the `pprof_config` contains the default [go pprof endpoint](https://pkg.go.dev/net/http/pprof) configuration:
+
+```river
+profiling_config {
+  pprof_config = {
+    memory = {
+      path = "/debug/pprof/allocs",
+      enabled = true,
+    }
+    block = {
+      path = "/debug/pprof/block",
+      enabled = true,
+    }
+    goroutine = {
+      path = "/debug/pprof/goroutine",
+      enabled = true,
+    },
+    mutex = {
+      path = "/debug/pprof/mutex",
+      enabled = true,
+    },
+    process_cpu = {
+      path = "/debug/pprof/profile",
+      enabled = true,
+      delta = true,
+    },
+  }
+}
+```
+
+You can use the `pprof_config` to override the default configuration or add new pprof endpoints.
+
+the `PprofProfilingConfig` attribute contains the following attributes:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`enabled`                  | `boolean`     | Enable this profile type to be scraped. | False | no
+`path`               | `string` | the path to the profile type on the target. | | no
+`delta`               | `boolean` | Whether to scrape the  profile as a delta. When scraping a delta profile, the query params `seconds` is automatically added. | False | no
+
+## Exported fields
+
+`pprof.scrape` does not export any fields that can be referenced by other
+components.
+
+## Component health
+
+`pprof.scrape` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+`pprof.scrape` reports the status of the last scrape for each configured
+scrape job on the component's debug endpoint.
+
+## Debug metrics
+
+* `agent_pprof_fanout_latency` (histogram): Write latency for sending to direct and indirect components.
```

Comment:
I'm not sure if we have a convention around this, but maybe we should start one. Do we prefix all debug metrics coming from Flow components with `agent_`, or do we just use the component name with underscores?

Reply:
We discussed this offline, and for now we're going to continue the convention of naming metrics based on the component name, and without the `agent_` prefix, so that this would be called `phlare_scrape_fanout_latency` (assuming pprof.scrape is renamed to phlare.scrape). 

If there's a name collision in dashboards, we can use `and` to filter down the set of metrics to the applications we want:

```
go_goroutines{cluster="$cluster",namespace="$namespace",instance=~"$instance"}
and on(instance)
agent_build_info{cluster="$cluster",namespace="$namespace",instance=~"$instance"}
```

Reply:
This is resolved.